File: 3310-remove-methods-from-project/3310-remove-methods-from-project.py
from collections import deque
from typing import List

class Solution:
    def remainingMethods(self, n: int, k: int, invocations: List[List[int]]) -> List[int]:
        g = [[] for _ in range(n)]
        for u, v in invocations:
            g[u].append(v)

        vis = [0] * n
        q = deque([k])
        vis[k] = 1
        while q:
            u = q.popleft()
            for v in g[u]:
                if not vis[v]:
                    vis[v] = 1
                    q.append(v)

        # Check for external invocations
        for u, v in invocations:
            if not vis[u] and vis[v]:
                return list(range(n))

        return [i for i in range(n) if not vis[i]]

File: 3310-remove-methods-from-project/3310-remove-methods-from-project.rs
use std::collections::VecDeque;

impl Solution {
    pub fn remaining_methods(n: i32, k: i32, invocations: Vec<Vec<i32>>) -> Vec<i32> {
        let n = n as usize;
        let k = k as usize;
        let mut g = vec![vec![]; n];
        for e in &invocations {
            let u = e[0] as usize;
            let v = e[1] as usize;
            g[u].push(v);
        }

        let mut vis = vec![false; n];
        let mut q = VecDeque::new();
        q.push_back(k);
        vis[k] = true;
        while let Some(u) = q.pop_front() {
            for &v in &g[u] {
                if !vis[v] {
                    vis[v] = true;
                    q.push_back(v);
                }
            }
        }

        for e in &invocations {
            let u = e[0] as usize;
            let v = e[1] as usize;
            if !vis[u] && vis[v] {
                return (0..n as i32).collect();
            }
        }

        (0..n as i32)
            .filter(|&i| !vis[i as usize])
            .collect()
    }
}